lib/dataset-tools/augmentation/replace_parameters.ts

```diff
@@ -751,6 +755,11 @@ export default class ParameterReplacer {
             }
 
             if (!type.isNumeric()) {
+                if (
+                    (this._paramLangPack.isGoodPersonName(sampled) && type_val?.includes("person")) ||
+                    (this._paramLangPack.isGoodUserName(sampled) && type_val && type_val.includes("username"))
+                )
+                    return { sentenceValue: sampled, programValue: sampled };
```

Comment:
What are you trying to do here? What is this giant HACK? No.

Reply:
I want to be able to sample foreign words for usernames irrespective of the language (e.g. pick mehrad0711 even if the sentence is Farsi). I guess there's a better way to do it. What do you suggest?

Reply:
It's fine to check if the entity type is tt:username, because that's a special entity. For other types of entities, add an annotation to the parameter and check that.

Reply:
Also, please use camelCase for local variables in JS